web/modules/custom/dpl_dashboard/src/Plugin/Block/DashboardBlock.php

```diff
@@ -0,0 +1,144 @@
+<?php
+
+namespace Drupal\dpl_dashboard\Plugin\Block;
+
+use Drupal\Core\Block\BlockBase;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\dpl_react_apps\Controller\DplReactAppsController;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides user intermediate list.
+ *
+ * @Block(
+ *   id = "dpl_dashboard_block",
+ *   admin_label = "List user dashboard"
+ * )
+ */
+class DashboardBlock extends BlockBase implements ContainerFactoryPluginInterface {
+  /**
+   * Drupal config factory.
+   *
+   * @var \Drupal\Core\Config\ConfigFactoryInterface
+   */
+  private ConfigFactoryInterface $configFactory;
+
+  /**
+   * DashboardBlock constructor.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin ID for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $configFactory
+   *   Drupal config factory to get FBS and Publizon settings.
+   */
+  public function __construct(
+    array $configuration,
+    $plugin_id,
+    $plugin_definition,
+    ConfigFactoryInterface $configFactory
+  ) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->configuration = $configuration;
+    $this->configFactory = $configFactory;
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('config.factory'),
+    );
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * @return mixed[]
+   *   The app render array.
+   */
+  public function build(): array {
+    $dashboardSettings = $this->configFactory->get('dashboard.settings');
+    $generalSettings = $this->configFactory->get('dpl_library_agency.general_settings');
+
+    $data = [
+      // Config.
+      "page-size-desktop" => $dashboardSettings->get('page_size_desktop'),
+      "page-size-mobile" => $dashboardSettings->get('page_size_mobile'),
+      "threshold-config" => $this->configFactory->get('dpl_library_agency.general_settings')->get('threshold_config'),
+      // Urls.
+      'dpl-cms-base-url' => DplReactAppsController::dplCmsBaseUrl(),
+      'fees-page-url' => $generalSettings->get('fees_page_url'),
+      'intermediate-url' => $dashboardSettings->get(""),
+      'loans-not-overdue-url' => $dashboardSettings->get(""),
+      'loans-overdue-url' => $dashboardSettings->get(""),
+      'loans-soon-overdue-url' => $dashboardSettings->get(""),
+      'pay-owed-url' => $dashboardSettings->get(""),
+      'physical-loans-url' => $dashboardSettings->get(""),
+      'reservations-url' => $dashboardSettings->get(""),
```

Comment:
There are a lot of empty configuration getters here?

Reply:
added